src/overlays/actors/ovl_Obj_Toge/z_obj_toge.c

```diff
@@ -32,52 +35,280 @@ const ActorInit Obj_Toge_InitVars = {
+    thisx->home.rot.z = thisx->world.rot.z = thisx->shape.rot.z = 0;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+
+    if (OBJTOGE_GET_PATH(thisx) == 0xFF) {
+        Actor_MarkForDeath(thisx);
+        return;
+    }
+
+    path = &globalCtx->setupPathList[OBJTOGE_GET_PATH(thisx)];
+    if (path->count != 2) {
+        Actor_MarkForDeath(thisx);
+        return;
+    }
+
+    sp40 = Lib_SegmentedToVirtual(path->points);
+    Math_Vec3s_ToVec3f(&this->unk_198[0], &sp40[0]);
+    Math_Vec3s_ToVec3f(&this->unk_198[1], &sp40[1]);
+    Math_Vec3f_Copy(&thisx->world.pos, &this->unk_198[0]);
+    thisx->world.rot.y = Math_Vec3f_Yaw(&this->unk_198[0], &this->unk_198[1]);
+    this->unk_194 = 0;
+    thisx->speedXZ = 0.0f;
+
+    if (sp3E > 0) {
+        s16 sp36;
+
+        this->unk_1B4 = true;
+        this->unk_1B8 = (this->unk_198[0].x + this->unk_198[1].x) * 0.5f;
+        this->unk_1BC = (this->unk_198[0].z + this->unk_198[1].z) * 0.5f;
+
+        sp36 = Math_Vec3f_Yaw(&this->unk_198[0], &this->unk_198[1]);
+        this->unk_1C8 = Math_CosS(sp36);
+        this->unk_1CC = Math_SinS(sp36);
+        this->unk_1C0 = sp3E * 7.5f;
+        this->unk_1C4 = Math3D_Distance(&this->unk_198[0], &this->unk_198[1]) * 0.5f;
+    } else {
+        this->unk_1B4 = false;
+    }
+
+    Collider_SetCylinder(globalCtx, &this->collider, thisx, &sCylinderInit);
+    Collider_UpdateCylinder(thisx, &this->collider);
+    thisx->colChkInfo.mass = MASS_IMMOVABLE;
+
+    if (sp38 == 1) {
+        this->collider.dim.radius = 60;
+        this->collider.dim.height = 40;
+    }
+
+    if (this->unk_1B4) {
+        func_809A4804(this);
+    } else {
+        func_809A4744(this);
+    }
+}
+
+void ObjToge_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjToge* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_809A4744(ObjToge* this) {
+    this->actionFunc = func_809A477C;
+    this->unk_1B0 = 0;
+    this->unk_1B2 = D_809A4CFC[OBJTOGE_GET_3800(&this->actor)];
+}
+
+void func_809A477C(ObjToge* this, GlobalContext* globalCtx) {
+    if (this->unk_1B2 > 0) {
+        this->unk_1B2--;
+    }
+
+    if (this->unk_1B0 >= 0x1770) {
+        this->unk_1B0 = 0x1770;
+        func_809A488C(this);
+    } else if ((this->unk_1B0 > 0) || (this->unk_1B2 <= 0)) {
+        this->unk_1B0 += 200;
+    }
+
+    if (!this->unk_1B4) {
+        this->actor.shape.rot.y += this->unk_1B0;
+    }
+}
 
-extern ColliderCylinderInit D_809A4CB0;
-extern InitChainEntry D_809A4D14[];
+void func_809A4804(ObjToge* this) {
+    this->actionFunc = func_809A481C;
+    this->unk_1B0 = 0;
+}
 
-extern UNK_TYPE D_06001400;
+void func_809A481C(ObjToge* this, GlobalContext* globalCtx) {
+    if (this->unk_1B0 >= 0x1770) {
+        this->unk_1B0 = 0x1770;
+        func_809A488C(this);
+    } else if ((this->unk_1B0 > 0) || func_809A43EC(this, globalCtx)) {
+        this->unk_1B0 += 200;
```

Comment:
```suggestion
        this->unk_1B0 += 0xC8;
```
 since it's rotation-related.